streaming/base/stream.py

```diff
@@ -138,13 +140,54 @@ def __init__(self,
                 raise ValueError('`download_timeout` must be positive')
             self.download_timeout = download_timeout
 
+        self.local_directory_timeout = local_directory_timeout
+        if local_directory_timeout <= 0:
+            raise ValueError('`local_directory_timeout` must be positive')
+
         self.validate_hash = validate_hash
 
+        if local is None:
```

Comment:
There are two cases where `torch.distributed.init_process_group()` is created for a distributed call. If it is initialized before creating the `StreamingDataset` then this solution would work. However, if it is not initialized then Streaming initialize it [here](https://github.com/mosaicml/streaming/blob/main/streaming/base/dataset.py#L322) which is after Streams are created. I would recommend to move the `destroy_dist = maybe_init_dist()` line before [this](https://github.com/mosaicml/streaming/blob/main/streaming/base/dataset.py#L290).